File: Python/1047. Remove All Adjacent Duplicates In String.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        st = []
        for a in s:
            if not st:
                st.append(a)
            elif st[-1] == a:
                st.pop(-1)
            else:
                st.append(a)
        return ''.join(st)

File: C++/1047. Remove All Adjacent Duplicates In String.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        string st;
        for (char a : s) {
            if (st.empty()) st.push_back(a);
            else if (st.back() == a) st.pop_back();
            else st.push_back(a);
        }
        return st;
    }
};